src/libs/actions/IOU.ts

```diff
@@ -5393,7 +5400,203 @@ function deleteMoneyRequest(transactionID: string, reportAction: OnyxTypes.Repor
         updatedReportPreviewAction.childMoneyRequestCount = reportPreviewAction.childMoneyRequestCount - 1;
     }
 
-    // STEP 5: Build Onyx data
+    let urlToNavigateBack: ReturnType<typeof ROUTES.REPORT_WITH_ID.getRoute> | undefined;
+
+    // STEP 5: Calculate what is the url that we will navigate user back
+    // This depends on which page they are on and which resources were deleted
+    if (iouReport && isSingleTransactionView && shouldDeleteTransactionThread && !shouldDeleteIOUReport) {
+        // Pop the deleted report screen before navigating. This prevents navigating to the Concierge chat due to the missing report.
+        urlToNavigateBack = ROUTES.REPORT_WITH_ID.getRoute(iouReport.reportID);
+    }
+
+    if (iouReport?.chatReportID && shouldDeleteIOUReport) {
+        // Pop the deleted report screen before navigating. This prevents navigating to the Concierge chat due to the missing report.
+        urlToNavigateBack = ROUTES.REPORT_WITH_ID.getRoute(iouReport.chatReportID);
+    }
```

Comment:
I don't understand the comment about popping the deleted report screen before navigating. Where does that take place? It looks like we're just setting up a URL to navigate back to here. It would be more clear without a comment.

NAB: Also, the code in both conditions is the same except for the reportID, so can we just set up a `reportIDToNavigateBackTo` variable instead. then use it to create the URL?